File: Spring Boot version/src/main/java/Thalassa/Models/Vessel.java
package Thalassa.Models;

import com.arangodb.springframework.annotation.Document;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import org.springframework.data.annotation.Id;

@Document("Vessels")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Vessel {

    @Id
    private String id;
    private String imoNumber, name, flag, company, yearBuilt, dwt;
    private boolean underConstruction;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getImoNumber() {
        return imoNumber;
    }

    public void setImoNumber(String imoNumber) {
        this.imoNumber = imoNumber;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getFlag() {
        return flag;
    }

    public void setFlag(String flag) {
        this.flag = flag;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public String getYearBuilt() {
        return yearBuilt;
    }

    public void setYearBuilt(String yearBuilt) {
        this.yearBuilt = yearBuilt;
    }

    public String getDwt() {
        return dwt;
    }

    public void setDwt(String dwt) {
        this.dwt = dwt;
    }

    public boolean isUnderConstruction() {
        return underConstruction;
    }

    public void setUnderConstruction(boolean underConstruction) {
        this.underConstruction = underConstruction;
    }
}

File: ASP.NET Core version/Models/Vessel.cs
using ArangoDB.Client;
using Newtonsoft.Json;

namespace Thalassa.Models
{
    [CollectionProperty(CollectionName = "Vessels")]
    public class Vessel
    {
        [JsonProperty(PropertyName = "id", NullValueHandling = NullValueHandling.Ignore),
        DocumentProperty(Identifier = IdentifierType.Key)]
        public string Id { get; set; }
        
        [JsonProperty(PropertyName = "imoNumber", NullValueHandling = NullValueHandling.Ignore),
        DocumentProperty(PropertyName = "imoNumber")]
        public string ImoNumber { get; set; }
       
        [JsonProperty(PropertyName = "name", NullValueHandling = NullValueHandling.Ignore),
        DocumentProperty(PropertyName = "name")]
        public string Name { get; set; }
        
        [JsonProperty(PropertyName = "flag", NullValueHandling = NullValueHandling.Ignore),
        DocumentProperty(PropertyName = "flag")]
        public string Flag { get; set; }
       
        [JsonProperty(PropertyName = "company", NullValueHandling = NullValueHandling.Ignore),
        DocumentProperty(PropertyName = "company")]
        public string Company { get; set; }
        
        [JsonProperty(PropertyName = "yearBuilt", NullValueHandling = NullValueHandling.Ignore),
        DocumentProperty(PropertyName = "yearBuilt")]
        public string YearBuilt { get; set; }
       
        [JsonProperty(PropertyName = "dwt", NullValueHandling = NullValueHandling.Ignore),
        DocumentProperty(PropertyName = "dwt")]
        public string Dwt { get; set; }
        
        [JsonProperty(PropertyName = "underConstruction", NullValueHandling = NullValueHandling.Ignore),
         DocumentProperty(PropertyName = "underConstruction")]
        public bool UnderConstruction { get; set; }
    }
}